modules/rostests/apitests/ntdll/RtlGetNtProductType.c

```diff
@@ -68,11 +68,66 @@ ReturnNtProduct(PDWORD ProductNtType)
     return FALSE;
 }
 
+static
+BOOLEAN
+ChangeNtProductType(DWORD NtProductType)
+{
+    PWSTR ProductTypeString;
+    LONG Result;
+    HKEY Key;
+    DWORD dwSize;
+
+    if (NtProductType == NtProductWinNt)
+    {
+        ProductTypeString = L"WinNT";
+    }
+    else if (NtProductType == NtProductLanManNt)
+    {
+        ProductTypeString = L"LanmanNT";
+    }
+    else if (NtProductType == NtProductServer)
+    {
+        ProductTypeString = L"ServerNT";
+    }
+    else
+    {
+        ok(FALSE, "Passed invalid product type to CHangeNtProduct: %lu", NtProductType);
+    }
+
+    Result = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                           L"SYSTEM\\CurrentControlSet\\Control\\ProductOptions",
+                           0,
+                           KEY_QUERY_VALUE,
+                           &Key);
+    if (Result != ERROR_SUCCESS)
+    {
+        ok(FALSE, "RegOpenKeyExW failed with 0x%lx\n", Result);
+        return FALSE;
+    }
+ 
+    dwSize = ((DWORD)wcslen(ProductTypeString) + 1) * sizeof(WCHAR);
```

Comment:
```suggestion
    dwSize = (DWORD)((wcslen(ProductTypeString) + 1) * sizeof(WCHAR));
```
In your way the wcslen was DWORD converted, but the doubling via the sizeof could wrap the size value in case it would have exceeded the maximum value that could be contained in DWORD.
In the suggestion, the doubling is done on size_t value, then it's truncated down to DWORD.
(Pick whatever you prefer ^^)

Reply:
LOL. That makes absolutely no difference.